audio_modder.py

```diff
@@ -1780,188 +2110,270 @@ def write_patch(self):
             patch_file_reader.wwise_banks = {}
             patch_file_reader.wwise_streams = {}
             patch_file_reader.text_banks = {}
-            
+
             for key, value in self.file_reader.wwise_streams.items():
                 if value.content.modified:
                     patch_file_reader.wwise_streams[key] = value
-                    
+
             for key, value in self.file_reader.wwise_banks.items():
                 if value.modified:
                     patch_file_reader.wwise_banks[key] = value
-                    
+
             for key, value in self.file_reader.text_banks.items():
                 for string_id in value.string_ids:
-                    if self.file_reader.string_entries[value.language][string_id].modified:
+                    if self.file_reader.string_entries[value.language][
+                        string_id
+                    ].modified:
                         patch_file_reader.text_banks[key] = value
                         break
-     
+
             patch_file_reader.rebuild_headers()
             patch_file_reader.to_file(folder)
         else:
             print("Invalid folder selected, aborting save")
             return False
         return True
 
-    def load_wems(self): 
-        wems = filedialog.askopenfilenames(title="Choose .wem files to import")
-        
+    def load_wems(self, wems=None):
+        if wems == None:
+            wems = filedialog.askopenfilenames(
+                    title="Choose .wem files to import"
+                    )
+
         progress_window = ProgressWindow(title="Loading Files", max_progress=len(wems))
         progress_window.show()
-        
+
         for file in wems:
-            progress_window.set_text("Loading "+os.path.basename(file))
-            file_id = self.get_number_prefix(os.path.basename(file))
+            progress_window.set_text("Loading " + os.path.basename(file))
+            file_id: int | None = None 
+
+            # TO-DO: require more robust check
+            is_from_wwise = "WwiseProjects" in file and ".cache" in file
+            has_wwise_hash = "_" in file
+
+            if is_from_wwise and has_wwise_hash:
+                basename = os.path.basename(file).split("_")[0] + ".wem"
```

Comment:
I think this check is unnecessary, but it doesn't hurt anything. The get_number_prefix function can already retrieve the id 1622724751373655930 from the filenames 1622724751373655930_56CEF360.wem, 1622724751373655930.wem, or 1622724751373655930HellpodDrop.wem.